.github/workflows/sonar.yml

```diff
@@ -4,39 +4,45 @@ on:
   push:
     branches:
       - main
-      - user/markackert/sonar
   pull_request:
     branches:
-      - "**"
+      - '**'
+
+permissions: read-all
 
 jobs:
   sonar:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0 
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js LTS
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
 
-    - name: Use Node.js LTS
-      uses: actions/setup-node@v2
-      with:
-        node-version: 16
+      - name: Use NPM v8
+        id: npm8
+        run: npm install -g npm@^8
 
-    - name: Use NPM v8
-      id: npm8
-      run: npm install -g npm@^8
+      - name: Install Node Package Dependencies
+        run: npm ci
 
-    - name: Install Node Package Dependencies
-      run: npm ci
+      - name: Build
+        run: npm run build
 
-    - name: Build 
-      run: npm run build
+      - name: Unit Tests
+        run: npm run test:unit
 
-    - name: SonarCloud Scan
-      uses: SonarSource/sonarcloud-github-action@master
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  
-        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+      - name: Integration Tests
+        run: npm run test:integration
 
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
```

Comment:
I guess we are using `@master` because they don't provide a single `v1` label?
Their release process (merging to master) seems stable enough though. 😋